amqtt/mqtt/protocol/client_handler.py

```diff
@@ -35,12 +35,10 @@ async def start(self):
 
     async def stop(self):
         await super().stop()
-        if self._ping_task:
-            try:
-                self.logger.debug("Cancel ping task")
-                self._ping_task.cancel()
-            except BaseException:
-                pass
+        if self._ping_task and not self._ping_task.cancelled():
```

Comment:
I favor asking for permission over forgiveness but that leaves a lot of room for discussion, especially as I'm not entirely sure about implications. @FlorianLudwig what's your opinion?